jena-arq/src/main/java/org/apache/jena/sparql/engine/iterator/QueryIterLateral.java

```diff
@@ -291,18 +291,25 @@ public Op transform(OpTable opTable) {
 
             // By the assignment restriction, the binding only needs to be added to each row of the table.
             Table table = opTable.getTable();
-            // Table vars.
-            List<Var> vars = new ArrayList<>(table.getVars());
-            binding.vars().forEachRemaining(vars::add);
-            TableN table2 = new TableN(vars);
+
+            TableBuilder tableBuilder = TableFactory.builder();
```

Comment:
Based on the discussion, I now added a TableBuilder class for building immutable Tables.